File: schemaorg/sportsevent.go
package schemaorg

import "encoding/json"

// SportsEvent see : https://schema.org/SportsEvent
type SportsEvent struct {

typeContext

Event

// AwayTeam see : https://schema.org/awayTeam
// The away team in a sports event.
AwayTeam interface{} `json:"awayTeam"` // types : Person SportsTeam

// Competitor see : https://schema.org/competitor
// A competitor in a sports event.
Competitor interface{} `json:"competitor"` // types : Person SportsTeam

// HomeTeam see : https://schema.org/homeTeam
// The home team in a sports event.
HomeTeam interface{} `json:"homeTeam"` // types : Person SportsTeam

}

func (v *SportsEvent) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "SportsEvent"

	return json.Marshal(v)
}

File: schemaorgphp/classes/sports_event.php
<?php

class SportsEvent extends Event implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'SportsEvent';
	
	/**
	 * The away team in a sports event.
	 * see : https://schema.org/awayTeam
	 * @var \Person|\Person[]|\SportsTeam|\SportsTeam[]
	 */
	public var $away_team;
	
	/**
	 * A competitor in a sports event.
	 * see : https://schema.org/competitor
	 * @var \Person|\Person[]|\SportsTeam|\SportsTeam[]
	 */
	public var $competitor;
	
	/**
	 * The home team in a sports event.
	 * see : https://schema.org/homeTeam
	 * @var \Person|\Person[]|\SportsTeam|\SportsTeam[]
	 */
	public var $home_team;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'SportsEvent'
		);
		
		$serialized = so_json_serialize( $this->away_team );
		if ( ! empty( $serialized ) ) {
			$out['awayTeam'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->competitor );
		if ( ! empty( $serialized ) ) {
			$out['competitor'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->home_team );
		if ( ! empty( $serialized ) ) {
			$out['homeTeam'] = $serialized;
		}
		
		return $out;
	}
}
